cmd/insights/verify-rule.go

```diff
@@ -6,51 +6,89 @@ import (
 	"os"
 
 	"github.com/fairwindsops/insights-cli/pkg/rules"
+	"github.com/google/go-cmp/cmp"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"gopkg.in/yaml.v3"
 )
 
-var inputFile, outputFile string
+var automationRuleFile, actionItemFile, expectedActionItem, reportType, insightsContext string
 
 func init() {
-	verifyRuleCmd.PersistentFlags().StringVarP(&inputFile, "input", "i", "./input.yaml", "Input file")
-	verifyRuleCmd.PersistentFlags().StringVarP(&outputFile, "output", "o", "./output.yaml", "Output file")
-	rootCmd.AddCommand(verifyRuleCmd)
+	verifyRuleCmd.PersistentFlags().StringVarP(&automationRuleFile, "automation-rule-file", "r", "./rule.js", "Automation rule JS file path")
+	verifyRuleCmd.PersistentFlags().StringVarP(&actionItemFile, "action-item-file", "a", "./action-item.yaml", "Action Item file path")
+	verifyRuleCmd.PersistentFlags().StringVarP(&insightsContext, "insights-context", "t", "", "Insights context: [AdmissionController/Agent/CI/CD]")
+	verifyRuleCmd.PersistentFlags().StringVarP(&reportType, "report-type", "R", "", "Report type")
+	verifyRuleCmd.PersistentFlags().StringVarP(&expectedActionItem, "expected-action-item", "i", "", "Optional file containing the action item that the automation rule is expected to produce")
+	validateCmd.AddCommand(verifyRuleCmd)
 }
 
 var verifyRuleCmd = &cobra.Command{
-	Use:   "verify-rule",
-	Short: "Verifies a rule against Insights",
-	Long:  "Verifies a rule to provided the action item and makes sure it will work fine testing it against Insights",
+	Use:   "rule -t  <insights context> {-r <rule file> -a <action item file>} [-i <expected output file>]",
+	Short: "Validates an automation rule",
+	Long:  "Validates an automation rule by applying it against the specified action item",
 	Run: func(cmd *cobra.Command, args []string) {
-		var verifyRule rules.VerifyRule
+		var actionItem rules.ActionItem
 		org := configurationObject.Options.Organization
 		host := configurationObject.Options.Hostname
-		in, err := os.Open(inputFile)
+		aiInput, err := os.Open(actionItemFile)
 		if err != nil {
-			exitWithError(fmt.Sprintf("Error when trying to open file %s", inputFile), err)
+			exitWithError(fmt.Sprintf("Error when trying to open file %s", actionItemFile), err)
 		}
-		inputContents, err := io.ReadAll(in)
+		aiBytes, err := io.ReadAll(aiInput)
 		if err != nil {
-			exitWithError(fmt.Sprintf("Could not read file %s", inputFile), err)
+			exitWithError(fmt.Sprintf("Could not read file %s", actionItemFile), err)
 		}
-		err = yaml.Unmarshal(inputContents, &verifyRule)
+		err = yaml.Unmarshal(aiBytes, &actionItem)
 		if err != nil {
 			exitWithError("Could not parse input file", err)
 		}
-		result, err := rules.RunVerifyRule(org, insightsToken, host, verifyRule)
+
+		ruleInput, err := os.Open(automationRuleFile)
+		if err != nil {
+			exitWithError(fmt.Sprintf("Error when trying to open file %s", actionItemFile), err)
```

Comment:
This error refers to the incorrect (action item) file name, as does the next error below it.

Reply:
done